### DISCUSSION TITLE
程序提供的下载链接怎么通过 curl 或 wget 下载？

### DISCUSSION THREAD

**Discussion by wc7086** [spencerwooo/onedrive-vercel-index#349] [Q&A] [answered]
用 curl 和 wget 随便试了几个参数都不能下载，不知道该用什么参数进行下载。

**Comment by spencerwooo** [comment]
带有 `?raw` 参数的链接有一个 302 redirect 的过程，可能需要有相应的参数让 curl 或者 wget 跟随 redirect 链接。

**Comment by spencerwooo** [comment, marked as answer]
Here - [Download file with url redirection](https://stackoverflow.com/questions/20440706/download-file-with-url-redirection).

```bash
curl -L <DOWNLOAD_URL>
curl --location <DOWNLOAD_URL>
```

Such as:

```bash
curl -L "https://drive.spencerwoo.com/api?path=/%F0%9F%8D%A1%20Genshin%20PV/PV%20-%20Eula%20-%20%E4%BC%98%E8%8F%88/%E3%80%8A%E5%8E%9F%E7%A5%9E%E3%80%8B%E8%A7%92%E8%89%B2%E6%BC%94%E7%A4%BA-%E3%80%8C%E4%BC%98%E8%8F%88%EF%BC%9A%E9%97%AA%E7%81%BC%E7%9A%84%E7%83%9B%E5%85%89%E3%80%8D%20-%20003%20-%20%E8%8B%B1.mp4&raw=true" -o genshin-eula-pv.mp4
```

![CleanShot 2022-02-04 at 17 33 45](https://user-images.githubusercontent.com/32114380/152505756-75332cb6-4c3d-4e8a-bacf-4f6e0ab4a2f6.png)

**Reply by wc7086** [reply]
我是忘记加双引号了 : (
